### DISCUSSION TITLE
webview.start gui kwarg seems mostly redundant

### DISCUSSION THREAD

**Discussion by adamlwgriffiths** [r0x0r/pywebview#772] [General]
Looking at the code in [guilib.py](https://github.com/r0x0r/pywebview/blob/master/webview/guilib.py#L77), it seems that the `gui` kwarg is almost entirely supefluous with the exception of prioritising gtk/qt

The check on line 72 shows a large list of what values are acceptable.
But the per-platform checks below only honor qt/gtk on some platforms.

Is there something I'm not grokking?

**Comment by r0x0r** [comment]
Do you mean forced_gui kwarg? If so, its purpose is to force a specific renderer trumping over the default renderer choice. If the forced renderer is not available it defaults to what is available.

**Comment by adamlwgriffiths** [comment]
Starting with [an example](https://pywebview.flowrl.com/examples/cef.html)
1. A call is made to `webview.start(gui)`
2. Which in turn calls [`guilib.initialize(gui)`](https://github.com/r0x0r/pywebview/blob/2ea34fdac1bcda4ebf9d9cb2bf4157bebbcccd5c/webview/__init__.py#L122)
3. `gui` is assigned to `forced_gui` and [`forced_gui` is then validated against a list of acceptable gui options](https://github.com/r0x0r/pywebview/blob/master/webview/guilib.py#L75)
4. This then used per platform
4a. [Darwin](https://github.com/r0x0r/pywebview/blob/master/webview/guilib.py#L78) only honors `forced_gui` if its `qt`.
4b. [Linux](https://github.com/r0x0r/pywebview/blob/master/webview/guilib.py#L87) only honors `forced_gui` if its `qt`.
4c. [Windows ignores `forced_gui`](https://github.com/r0x0r/pywebview/blob/master/webview/guilib.py#L96)

Unless I've missed something, It looks like its really not doing much of anything, except putting `qt` at the front of the list of `[qt, gtk]` on Darwin/Linux.
All other options are ignored.

**Comment by r0x0r** [comment]
Yes you are right that is used only for forcing QT. However if you look at winforms.py the same parameter is used for choosing a renderer, even though WinForms is used in any case.

**Comment by adamlwgriffiths** [comment]
Ah ok, I see what's going on.
[`forced_gui_`](https://github.com/r0x0r/pywebview/blob/master/webview/guilib.py#L9) is a global, that the [other platforms then reach into to use.](https://github.com/r0x0r/pywebview/blob/master/webview/platforms/winforms.py#L111)
I didn't see that connection there.
Thanks for pointing me in the right direction.
